core/src/test/scala/org/locationtech/rasterframes/functions/TileFunctionsSpec.scala

```diff
@@ -516,10 +519,58 @@ class TileFunctionsSpec extends TestEnvironment with RasterMatchers {
       val expr = df.select(rf_render_png($"red", "HeatmapBlueToYellowToRedSpectrum"))
 
       val pngData = expr.first()
-
+      // Transform it back into a raster.
       val image = ImageIO.read(new ByteArrayInputStream(pngData))
       image.getWidth should be(red.cols)
       image.getHeight should be(red.rows)
     }
+
+    it("should distribute classes across color ramp") {
```

Comment:
this is really cool. we add a few edge cases along this line:

 - tile is full of nodata
 - tile is nodata and 1 other value
 - tile is only 1 value